### PR TITLE
http: removing the http1 and http2 connection pools

### PR DESCRIPTION
**Pull request by alyssawilk**
This is a refactor to make #13922 safe long term.

Basically the HTTP/1 and HTTP/2 logic used to be split between the ActiveClient, and the ConnectionPool classes.
Prior refactors moved all the logic from the pool to the client, so that the new MixedConnectionPool could have HTTP/1 or HTTP/2 clients, with all the logic contained in the ActiveClient.  This does away with the pool classes entirely, so folks don't add any logic back.

This could go one of two ways.  We can keep the FixedConnectionPool and MixedConnectionPool long term, or we can have a single merged pool with if (alpn) {alpn_logic} else {fixed_logic}.  I don't have strong feelings, but felt the split was a bit cleaner.

Risk Level: Medium
Testing: n/a (pure refactor)
Docs Changes: n/a
Release Notes: n/a
Part of  #3431

### CODE REVIEW

**Comment by alyssawilk** [comment]
cc @snowp and @mattklein123  (it's the last conn pool refactor, I swear! :-P )

File: source/common/http/conn_pool_base.h [hunk 1]
```diff
@@ -111,10 +121,43 @@ class ActiveClient : public Envoy::ConnectionPool::ActiveClient {
...
+    return codec_fn_(data, this);
+  }
+
+  Envoy::ConnectionPool::ActiveClientPtr instantiateActiveClient() override {
+    return client_fn_(this);
+  }
+
+protected:
+  CreateCodecFn codec_fn_;
+  CreateClientFn client_fn_;
```
Comment (mattklein123) [inline source/common/http/conn_pool_base.h:158]: nit: const

**Review by mattklein123 — CHANGES_REQUESTED** [review]
Nice, just a small nit.

/wait

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**